api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentGroupController.java

```diff
@@ -327,6 +330,63 @@ public PaymentGroupController(PaymentGroupService paymentGroupService, PaymentGr
         return new ResponseEntity<>(paymentDtoMapper.toBulkScanPaymentDto(newPayment, paymentGroupReference), HttpStatus.CREATED);
     }
 
+
+    @ApiOperation(value = "Create telephony card payment in Payment Group", notes = "Create telephony card payment in Payment Group")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Payment created"),
+        @ApiResponse(code = 400, message = "Payment creation failed"),
+        @ApiResponse(code = 422, message = "Invalid or missing attribute")
+    })
+    @PostMapping(value = "/payment-groups/{payment-group-reference}/telephony-card-payments")
+    @ResponseBody
+    @Transactional
+    public ResponseEntity<TelephonyCardPaymentsResponse> createTelephonyCardPayment(
+        @RequestHeader(value = "return-url") String returnURL,
+        @PathVariable("payment-group-reference") String paymentGroupReference,
+        @Valid @RequestBody TelephonyCardPaymentsRequest telephonyCardPaymentsRequest) throws CheckDigitException, MethodNotSupportedException {
+
+        boolean antennaFeature = featureToggler.getBooleanValue("pci-pal-antenna-feature",false);
+        LOG.info("Feature Flag Value in CardPaymentController : {}", antennaFeature);
+        if(antennaFeature) {
+            LOG.info("Inside Telephony check!!!");
+            TelephonyProviderAuthorisationResponse telephonyProviderAuthorisationResponse = pciPalPaymentService.getPaymentProviderAutorisationTokens();
+            PaymentServiceRequest paymentServiceRequest = PaymentServiceRequest.paymentServiceRequestWith()
+                .paymentGroupReference(paymentGroupReference)
+                .paymentReference(referenceUtil.getNext("RC"))
+                .ccdCaseNumber(telephonyCardPaymentsRequest.getCcdCaseNumber())
+                .currency(telephonyCardPaymentsRequest.getCurrency().getCode())
+                .siteId(telephonyCardPaymentsRequest.getSiteId())
+                .serviceType(telephonyCardPaymentsRequest.getService().getName())
+                .amount(telephonyCardPaymentsRequest.getAmount())
+                .channel(PaymentChannels.TELEPHONY.getCode())
+                .provider(PaymentProviders.PCIPAL.getCode())
+                .build();
+            LOG.info("Access Token Value in CardPaymentController : {}", telephonyProviderAuthorisationResponse.getAccessToken());
+            LOG.info("Refresh Token Value in CardPaymentController : {}", telephonyProviderAuthorisationResponse.getRefreshToken());
```

Comment:
We should not log these tokens at info level, they will be logged in Prod, use log.debug instead

Reply:
This is a temporary logger for debugging purpose in demo and PR. We will remove it while pushing to production.